pkg/controllers/secretstore/common.go

```diff
@@ -51,12 +62,24 @@ type Opts struct {
 	RequeueInterval time.Duration
 }
 
-func reconcile(ctx context.Context, req ctrl.Request, ss esapi.GenericStore, cl client.Client, log logr.Logger, opts Opts) (ctrl.Result, error) {
+func reconcile(ctx context.Context, req ctrl.Request, ss esapi.GenericStore, cl client.Client, isPushSecretEnable bool, log logr.Logger, opts Opts) (ctrl.Result, error) {
 	if !ShouldProcessStore(ss, opts.ControllerClass) {
 		log.V(1).Info("skip store")
 		return ctrl.Result{}, nil
 	}
 
+	finalizersUpdated, err := handleFinalizer(ctx, cl, ss, log, isPushSecretEnable)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if finalizersUpdated {
+		log.V(1).Info("updating resource with finalizer changes")
+		if err := cl.Update(ctx, ss); err != nil {
```

Comment:
after calling update, you should not keep working with the object but you should requeue no?